autogen/main/networks.tf

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+{{ autogeneration_note }}
+
+data "google_compute_network" "gke_network" {
```

Comment:
Note this might raise some issues with networks created inline. Could we make these data sources conditional on the firewall_enabled var?

Reply:
Will look into doing that.

Reply:
I think the regular count conditional approach should work on datasources too

Reply:
I think this is the only thing blocking merge now.

Reply:
Should be fixed in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/470/commits/8ffa5e8326de3cab6db7cbc11c8019e4e1ecc9df 
Also removed the top level network datasource since it seems unused

Reply:
Plus https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/470/commits/ceef39536910631b11db8af544e06eaeb437c019 to fix attribute errors, not pretty but it works/keeps it all conditional.

```tf
Error: Unsupported attribute

  on .terraform/modules/blue_gke.gke/main.tf line 75, in locals:
  75:   cluster_alias_ranges_cidr = { for range in toset(data.google_compute_subnetwork.gke_subnetwork.*.secondary_ip_range) : range.range_name => range.ip_cidr_range }
```